### PR TITLE
Plugins api docs

### PR DESCRIPTION
**Pull request by pksjce**

### CODE REVIEW

File: content/pluginsapi/index.md [hunk 1]
```diff
@@ -3,12 +3,44 @@ title: Plugins API
 sort: 1
 ---
 
+webpack provides flexible and powerful customization api in the form of plugins. Using plugins, we can plug functionality into webpack. Additionaly, webpack provides lifecycle hooks into which plugins can be registered. At each of these lifecycle points, webpack will run all of the registered plugins and provide them with the current state of the webpack compilation.
```
Comment (SpaceK33z) [inline content/pluginsapi/index.md]: s/Additionaly/Additionally/

File: content/concepts/plugins.md [hunk 2]
```diff
@@ -23,7 +22,7 @@ function ConsoleLogOnBuildWebpackPlugin() {
 };
 
 ConsoleLogOnBuildWebpackPlugin.prototype.apply = function(compiler) {
-  compiler.plugin('run', function(compiler, callback) {
+  compiler.plugin('run', function(compilation, callback) {
```
Comment (sokra) [inline content/concepts/plugins.md]: `compiler.plugin('run', function(compiler, callback) {`

File: content/pluginsapi/compiler.md [hunk 3]
```diff
@@ -3,14 +3,78 @@ title: Compiler
...
+    console.log('Compiler has finished execution.');
+    /* Do something to show the stats */
+};
+
+// call run on the compiler along with the callback
+compiler.run(callback);
+```
+
+The `Compiler` is what we call a `Tapable` instance. By this, we mean that it mixes in `Tapable` class to imbibe functionality to register and call plugins on itself.
+All user facing plugins are first registered on the `Compiler`.
```
Comment (sokra) [inline content/pluginsapi/compiler.md]: That's not totally true. There is a `resolve.plugins` option which allows to register resolving plugins directly.

Comment (sokra) [inline content/pluginsapi/compiler.md]: It's not via the plugin API. You can pass it via configuration:

``` js
module.exports = {
  resolve: {
    plugins: [
      new ResolvingPlugin({...})
    ]
  }
}
```

Comment (pksjce) [inline content/pluginsapi/compiler.md]: Oh I did not know about this. Need to write it in [resolver config doc](https://webpack.js.org/configuration/resolve/#resolve-plugins). So how do you think I should modify this line?

Comment (sokra) [inline content/pluginsapi/compiler.md]: All -> Most

Comment (pksjce) [inline content/pluginsapi/compiler.md]: Cool! Will do it. You were talking about [this in ResolverFactory](https://github.com/webpack/enhanced-resolve/blob/master/lib/ResolverFactory.js) ?

**Comment by pksjce** [comment]
Shall we freeze this here? I do have more content cooking, for which I can raise new PRs?

**Comment by bebraw** [comment]
Yeah, separate PRs will work.

**Comment by pksjce** [comment]
Thanks!

**Pull request merged**